In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [7]:
cricket = pd.read_csv("cricket_data.csv")
cricket.head()

Year      Player_Name Matches_Batted Not_Outs Runs_Scored Highest_Score  \
0  2024  Ruturaj Gaikwad              2        0          61            46   
1  2023  Ruturaj Gaikwad             16        1         590            92   
2  2022  Ruturaj Gaikwad             14        0         368            99   
3  2021  Ruturaj Gaikwad             16        2         635          101*   
4  2020  Ruturaj Gaikwad              6        2         204            72   

  Batting_Average Balls_Faced Batting_Strike_Rate Centuries  ...  \
0            30.5          51              119.61         0  ...   
1           42.14         400               147.5         0  ...   
2           26.29         291              126.46         0  ...   
3           45.35         466              136.26         1  ...   
4              51         169              120.71         0  ...   

  Matches_Bowled Balls_Bowled Runs_Conceded Wickets_Taken Best_Bowling_Match  \
0              2            0             0             0                  0   
1             16            0             0             0                  0   
2             14            0             0             0                  0   
3             16            0             0             0                  0   
4              6            0             0             0                  0   

  Bowling_Average Economy_Rate Bowling_Strike_Rate Four_Wicket_Hauls  \
0               0            0                   0                 0   
1               0            0                   0                 0   
2               0            0                   0                 0   
3               0            0                   0                 0   
4               0            0                   0                 0   

  Five_Wicket_Hauls  
0                 0  
1                 0  
2                 0  
3                 0  
4                 0  

[5 rows x 25 columns]

In [8]:
cricket.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1172 entries, 0 to 1171
Data columns (total 25 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Year                 1172 non-null   object
 1   Player_Name          1172 non-null   object
 2   Matches_Batted       1172 non-null   object
 3   Not_Outs             1172 non-null   object
 4   Runs_Scored          1172 non-null   object
 5   Highest_Score        1172 non-null   object
 6   Batting_Average      1172 non-null   object
 7   Balls_Faced          1172 non-null   object
 8   Batting_Strike_Rate  1172 non-null   object
 9   Centuries            1172 non-null   object
 10  Half_Centuries       1172 non-null   object
 11  Fours                1172 non-null   object
 12  Sixes                1172 non-null   object
 13  Catches_Taken        1172 non-null   object
 14  Stumpings            1172 non-null   object
 15  Matches_Bowled       1172 non-null   object
 16  Balls_

We can observe that the data set here is complete and there are no NaN(Not a Number)/ Missing Values. This saves a lot of time in feature engineering.

In [9]:
cricket.sample()

Year     Player_Name Matches_Batted Not_Outs Runs_Scored Highest_Score  \
995  2021  Mahipal Lomror              4        1          94            43   

    Batting_Average Balls_Faced Batting_Strike_Rate Centuries  ...  \
995           31.33          73              128.76         0  ...   

    Matches_Bowled Balls_Bowled Runs_Conceded Wickets_Taken  \
995              4           42            47             1   

    Best_Bowling_Match Bowling_Average Economy_Rate Bowling_Strike_Rate  \
995               1/22           47.00         6.71               42.00   

    Four_Wicket_Hauls Five_Wicket_Hauls  
995                 0                 0  

[1 rows x 25 columns]

In [16]:
cricket['Year'].value_counts()

Year
2023        162
2022        141
2024        121
2021        110
2020        100
2019         92
2018         74
2017         64
2016         55
2015         47
2014         42
No stats     42
2013         33
2012         25
2011         21
2010         16
2009         15
2008         12
Name: count, dtype: int64

We can observe that there are 141 values from the year 2022. And according to required statement, we are supposed to filter out data from this specific year alone. So let us filter using the year.

In [20]:
cricket[cricket['Year']=='2022']

Year       Player_Name Matches_Batted Not_Outs Runs_Scored  \
2     2022   Ruturaj Gaikwad             14        0         368   
6     2022      Devon Conway              7        1         252   
9     2022          MS Dhoni             14        6         232   
26    2022    Ajinkya Rahane              7        0         133   
45    2022   Ravindra Jadeja             10        4         116   
...    ...               ...            ...      ...         ...   
1141  2022  Fazalhaq Farooqi              3        1           2   
1143  2022    Jaydev Unadkat              5        2          59   
1158  2022       T Natarajan             11        0           0   
1163  2022       Umran Malik             14        3           4   
1167  2022   Mayank Markande              2        0           0   

     Highest_Score Batting_Average Balls_Faced Batting_Strike_Rate Centuries  \
2               99           26.29         291              126.46         0   
6               87              42         173              145.66         0   
9              50*           33.14         188              123.40         0   
26              44              19         128              103.91         0   
45             26*           19.33          98              118.37         0   
...            ...             ...         ...                 ...       ...   
1141            2*               0           8                  25         0   
1143           19*           19.67          37              159.46         0   
1158             0               0           0                   0         0   
1163            3*               4           8                  50         0   
1167             0               0           0                   0         0   

      ... Matches_Bowled Balls_Bowled Runs_Conceded Wickets_Taken  \
2     ...             14            0             0             0   
6     ...              7            0             0             0   
9     ...             14            0             0             0   
26    ...              7            0             0             0   
45    ...             10          198           248             5   
...   ...            ...          ...           ...           ...   
1141  ...              3           72           110             2   
1143  ...              5          120           190             6   
1158  ...             11          258           406            18   
1163  ...             14          295           444            22   
1167  ...              2           42            57             1   

     Best_Bowling_Match Bowling_Average Economy_Rate Bowling_Strike_Rate  \
2                     0               0            0                   0   
6                     0               0            0                   0   
9                     0               0            0                   0   
26                    0               0            0                   0   
45                 3/39           49.60         7.52               39.60   
...                 ...             ...          ...                 ...   
1141               2/32              55         9.17                  36   
1143               2/32           31.67          9.5               20.00   
1158               3/10           22.56         9.44               14.33   
1163               5/25           20.18         9.03               13.41   
1167               1/26           57.00         8.14               42.00   

     Four_Wicket_Hauls Five_Wicket_Hauls  
2                    0                 0  
6                    0                 0  
9                    0                 0  
26                   0                 0  
45                   0                 0  
...                ...               ...  
1141                 0                 0  
1143                 0                 0  
1158                 0                 0  
1163                 1                 1  
1167         

In [22]:
cricket[cricket['Year']=='2022'].value_counts().sum()

np.int64(141)

And this way we are easily able to filter out the required data. This is in the format of a pandas dataframe. To convert it into a csv file. But first lets store this data frame in a variable.

In [23]:
cricket_new = cricket[cricket['Year']=='2022']

In [27]:
cricket_new.to_csv("cricket_newdata.csv", index = False)